packages/menu-button/src/index.tsx

```diff
@@ -900,30 +900,28 @@ export const MenuPopover = forwardRef<any, MenuPopoverProps>(
     const ref = useForkedRef(popoverRef, forwardedRef);
 
     useEffect(() => {
-      function listener(event: MouseEvent) {
+      function listener(event: MouseEvent | TouchEvent) {
         if (buttonClickedRef.current) {
           buttonClickedRef.current = false;
         } else {
           // We on want to close only if focus rests outside the menu
-          if (isExpanded && popoverRef.current) {
-            if (!popoverRef.current.contains(event.target as Element)) {
-              dispatch({ type: CLOSE_MENU, payload: { buttonRef } });
-            }
+          if (
+            !popoverRef.current ||
+            popoverRef.current.contains(event.target as Element)
+          ) {
+            return;
           }
+
+          dispatch({ type: CLOSE_MENU, payload: { buttonRef } });
         }
       }
-      window.addEventListener("mousedown", listener);
+      document.addEventListener("mousedown", listener);
+      document.addEventListener("touchstart", listener);
       return () => {
-        window.removeEventListener("mousedown", listener);
+        document.removeEventListener("mousedown", listener);
+        document.removeEventListener("touchstart", listener);
```

Comment:
The addition of `touchstart` events here means that the popover is closed as soon as you begin scrolling on a touch device. Is this desired?

Reply:
I don't think that's a problem. As a user, if you are scrolling you are not really paying attention to the popover anymore right?

Reply:
I don't think that's a safe assumption, especially if the menu items overflow the screen.

Reply:
@cpmsmith You raise a good point. I'm not sure what the answer is but I'll revert this change and open a new issue to consider this case separately.

Reply:
I think it should be fine to just leave it with `mousedown`, no? Mobile browsers will fire it on any tap, but not scrolling, which is what I'd expect.

Reply:
There are a few menu button type interactions on iOS that do not close the menu on immediate touch, but they do close it when you start scrolling. There may be cases where you do and don't want this on a web UI, so it may be worth exploring.